src/App.tsx

```diff
@@ -1,9 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+
+  const visibleMovies = [...moviesFromServer].filter(movie => {
+    const movieTitleLowerCase = movie.title.toLowerCase();
+    const movieDescriptionLowerCase = movie.description.toLowerCase();
+    const queryLowerCase = query.toLowerCase();
+
+    return movieTitleLowerCase.includes(queryLowerCase)
+     || movieDescriptionLowerCase.includes(queryLowerCase);
+  });
```

Comment:
Agree, you shouldn't make a copy. Also, create function to filter movies and use here `useMemo`

Reply:
Ok